Source/SwiftLintFramework/Models/ExcludedRegexExpression.swift

```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+/// Represents regex from `exclude` option in `NameConfiguration`.
+/// Using NSRegularExpression causes failure on equality check between two `NameConfiguration`s.
+/// This class compares pattern only when checking its equality
+public final class ExcludedRegexExpression: NSObject {
```

Comment:
Could you please explain why this class is necessary? There also exist other configurations with `NSRegularExpression`-typed fields.

Reply:
I found `RuleConfigurationTests.testNameConfigurationSetsCorrectly` fails on CI machine when using `NSRegularExpression` type in `NameConfiguration`. ( It succeeds on my local machine )

The test compares two `NameConfiguration` with XCTAssertEqual and those configurations holds two different `NSRegularExpression` instances and causes the test to fail.

Reply:
I found `RuleConfigurationTests.testNameConfigurationSetsCorrectly` fails on CI machine when using `NSRegularExpression` type in `NameConfiguration`. ( It succeeds on my local machine )

The test compares two `NameConfiguration` with XCTAssertEqual and those configurations holds two different `NSRegularExpression` instances and causes the test to fail.